examples/http-storage/index.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright 2016-2019, Pulumi Corporation.  All rights reserved.
+
+import * as azure from '@pulumi/azure';
+import { HttpRequest } from '@pulumi/azure/appservice';
+
+const resourceGroup = new azure.core.ResourceGroup('example', { location: 'West US 2' });
+
+const storageAccount = new azure.storage.Account("storage", {
+  resourceGroupName: resourceGroup.name,
+  accountReplicationType: "LRS",
+  accountTier: "Standard",
+});
+
+const args: azure.appservice.HttpEventSubscriptionArgs = {
+  resourceGroup,
+  callback: async (context, req: HttpRequest) => {
+
+    context.bindings.person = [];
+    context.bindings.person.push({
+      PartitionKey: req.query.email,
+      RowKey: req.query.email,
+      Name: req.query.name
+    });
+
+    return {
+      status: 200,
+      body: `Hello ${req.query.name}!`
+    };
+  },
+  methods: ["GET"],
+  account: storageAccount
+};
+
+storageAccount.bindTableOutput(args, "person", "Person");
+
+const greeting = new azure.appservice.HttpEventSubscription('greeting', args);
```

Comment:
I would try to keep the usual shape of API with args inline in the constructor call. The 3-step process feels a bit alien here. We should be able to do something like

``` ts
const greeting = new azure.appservice.HttpEventSubscription('greeting', {
  resourceGroup,
  callback: ...,
  methods: ["GET"],
  account: storageAccount,
  outputBindings: [storageAccount.tableOutput("person", "Person")]
});
```

with merging of app settings happening inside our implementation classes.

By the way, why not use an `azure.storage.Table` instance?

Reply:
What do you mean with 
>  inside our implementation classes

Do you mean that `HttpEventSubscription` or its base class implements logic that uses information from the `outputBindings` array of the args to modify both the app settings as well as the bindings used for the function? Sounds doable and I agree that this looks cleaner.

Reply:
Yes, one of those (it's also changing a bit in the other PR which will hopefully be merged very soon).
I'll play with these ideas a bit more to shape the ideas more specifically, sorry for being vague.

Reply:
Since issue #232 is still open - is this now the way we want to go forward with this? If yes -> I'll refactor my PR to fit your proposal in a few days.

Reply:
Please check my latest comment in the issue https://github.com/pulumi/pulumi-azure/issues/232#issuecomment-501274842

Reply:
and regarding why I dont use `azure.storage.Table` : from the table, I dont have access to the connection string, but from the storage account, I do.